### ISSUE TITLE
Window creation after re-initialization blocks indefinitely

### ISSUE THREAD

**Issue by durswd** [glfw/glfw#1439]
I repeat glfwInit - CreateWindow - glfwTerminate.
The program was froze when glfwCreateWindow is called twice.

My environment macOS High Sierra.

```c
for(int i = 0; i < 2; i++)
{
  if (!glfwInit())
    exit(EXIT_FAILURE);
  GLFWwindow* window;
  int running = GLFW_TRUE;
  window = glfwCreateWindow(640, 480, "Empty Event Test", NULL, NULL);
  if (!window) {
      glfwTerminate();
      exit(EXIT_FAILURE);
  }

  glfwMakeContextCurrent(window);
  while (running) {  
    glfwSwapBuffers(window);
    glfwPollEvents();

    if (glfwWindowShouldClose(window))
      running = GLFW_FALSE;
  }
  glfwHideWindow(window);
  glfwDestroyWindow(window);
  glfwTerminate();
}
```

**Comment by elmindreda** [comment]
That is definitely a bug, thank you!

**Comment by elmindreda** [comment]
Note that you shouldn't have to re-initialize GLFW unless the structure of your project somehow requires that. Doing it should still work, of course.

**Comment by elmindreda** [comment]
This should be fixed now with 2fbb560eb72fea695687d33066c91a329d6d0465.

**Comment by durswd** [comment]
Thank you. My project requires that.
